next.js - missing dependency - can't run production build

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
Missing dependency `@nrwl/workspace` used in `next.config.js` inside production build 

I think it's because `@nrwl/next` has `@nrwl/workspace` as peer dependency: 

https://github.com/nrwl/nx/blob/e63c3d162bd2b61c1991f42ac6bde012bc4ff658/packages/next/package.json#L31-L34

## Expected Behavior
All dependencies should be defined in package.json

## Steps to Reproduce
```
npx create-nx-workspace
cd ./my-org
npx nx build
mv ./dist/apps/app/ ../production-build
cd ../production-build
npm install
npm start
```

`package.json`

```
{
  "name": "app",
  "version": "0.0.1",
  "scripts": {
    "start": "next start"
  },
  "dependencies": {
    "react": "17.0.1",
    "next": "10.0.1",
    "react-dom": "17.0.1"
  },
  "devDependencies": {
    "@nrwl/next": "11.0.2",
    "@testing-library/react": "11.1.2"
  }
}
```

`next.config.js`

```
const withNx = require('@nrwl/next/plugins/with-nx');

module.exports = withNx({});
```

### Failure Logs
```
> app@0.0.1 start C:\Users\rafal\Projects\temp\production-build
> next start

Error: Cannot find module '@nrwl/workspace/src/utils/app-root'
Require stack:
- C:\Users\rafal\Projects\temp\production-build\node_modules\@nrwl\next\plugins\with-nx.js
- C:\Users\rafal\Projects\temp\production-build\next.config.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\next-server\server\config.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\next-server\server\next-server.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\server\next.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\server\lib\start-server.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\cli\next-start.js
- C:\Users\rafal\Projects\temp\production-build\node_modules\next\dist\bin\next
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\Users\rafal\Projects\temp\production-build\node_modules\@nrwl\next\plugins\with-nx.js:2:25)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\@nrwl\\next\\plugins\\with-nx.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\next.config.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\next-server\\server\\config.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\next-server\\server\\next-server.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\server\\next.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\server\\lib\\start-server.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\cli\\next-start.js',
    'C:\\Users\\rafal\\Projects\\temp\\production-build\\node_modules\\next\\dist\\bin\\next'
  ]
}
```

### Environment
```
  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.0.2
  @nrwl/cypress : 11.0.2
  @nrwl/devkit : 11.0.2
  @nrwl/eslint-plugin-nx : 11.0.2
  @nrwl/express : Not Found
  @nrwl/jest : 11.0.2
  @nrwl/linter : 11.0.2
  @nrwl/nest : Not Found
  @nrwl/next : 11.0.2
  @nrwl/node : Not Found
  @nrwl/react : 11.0.2
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.0.2
  @nrwl/web : 11.0.2
  @nrwl/workspace : 11.0.2
  typescript : 4.0.5
```

--- a/packages/next/package.json
+++ b/packages/next/package.json
@@ -30,7 +30,7 @@
   },
   "peerDependencies": {
     "@nrwl/workspace": "*",
-    "next": "^9.3.3"
+    "next": "^10.0.3"
   },
   "dependencies": {
     "@nrwl/react": "*",

--- a/packages/next/src/builders/build/lib/create-package-json.ts
+++ b/packages/next/src/builders/build/lib/create-package-json.ts
@@ -15,8 +15,12 @@ function getProjectDeps(context: BuilderContext, rootPackageJson: any) {
   const depNames = deps
     .map((d) => d.node)
     .filter((node) => node.type === 'npm')
-    .map((node) => node.data.packageName);
-  const dependencies = depNames
+    .map((node) => node.data.packageName)
+    // Need to make sure @nrwl/workspace is installed
+    // It is only a peer dependency of @nrwl/next so does not get installed automatically
+    // See: https://github.com/nrwl/nx/issues/4336
+    .concat('@nrwl/workspace');
+  const dependencies: string[] = depNames
     .filter((packageName) => packageName in rootPackageJson.dependencies)
     .reduce((deps, pkgName) => {
       return { ...deps, [pkgName]: rootPackageJson.dependencies[pkgName] };

